app.js

```diff
@@ -0,0 +1,198 @@
+require("dotenv").config();
+const express = require("express");
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const bodyParser = require("body-parser");
+const randomString = require("crypto-random-string");
+const Register = require("./models/userRegistraion");
+const AccessToken = require("./models/accessToken");
+const Address = require("./models/address");
+
+const router = express.Router();
+const app = express();
+
+mongoose
+  .connect(process.env.DATABASE_URI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    reconnectTries: 30,
+    reconnectInterval: 500,
+    poolSize: 10,
+    bufferMaxEntries: 0,
+    autoIndex: true,
+    useCreateIndex: true,
+    useFindAndModify: false,
+  })
+  .then(() => {
+    console.log("Link established to database");
+  })
+  .catch((err) => {
+    console.log("No link to database.", err);
+  });
+
+router.post("/user/register", async (req, res) => {
+  console.log(req);
+  console.log(req.body);
+  try {
+    let {
+      firstName,
+      secondName,
+      userName,
+      password,
+      confirmPassword,
+      email,
+    } = req.body;
+    let validatedPassword;
+    if (password === confirmPassword) {
+      validatedPassword = await bcrypt.hash(password, 10);
+    } else {
+      throw "password did not match";
+    }
+    let userRecord = await Register.findOne({
+      $or: [{ email: email }, { userName: userName }],
+    });
+    if (userRecord) {
+      throw "username or email is already exist";
+    }
+
+    let userData = {
+      firstName: firstName,
+      secondName: secondName,
+      userName: userName,
+      password: validatedPassword,
+      email: email,
+    };
+    let user = new Register(userData);
+    await user.save();
+
+    res.status(200).send("registered succefully");
+  } catch (error) {
+    res.status(500).send({
+      error: error,
+    });
+  }
+});
+
+router.post("/user/login/:username/:password", async (req, res) => {
```

Comment:
don't use params to send the data of credentials (like login etc)
even you are using the POST method so use the request body